pkg/controller/migcluster/validation.go

```diff
@@ -216,6 +226,47 @@ func (r ReconcileMigCluster) testConnection(cluster *migapi.MigCluster) error {
 	return nil
 }
 
+// Validate the Exposed registry route
+func (r ReconcileMigCluster) validateRegistryRoute(cluster *migapi.MigCluster) error {
+
+	if cluster.Status.HasCriticalCondition() {
+		return nil
+	}
+
+	if cluster.Spec.ExposedRegistryPath != "" {
+		url := "https://" + cluster.Spec.ExposedRegistryPath + "/v2/"
+		restConfig, err := cluster.BuildRestConfig(r.Client)
+		token := restConfig.BearerToken
+
+		tr := &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		}
+		client := &http.Client{Transport: tr}
+
+		req, err := http.NewRequest("GET", url, nil)
+
+		if err != nil {
+			return liberr.Wrap(err)
+		}
+
+		req.Header.Set("Authorization", "bearer "+token)
+
+		res, err := client.Do(req)
+		if err != nil {
+			cluster.Status.SetCondition(migapi.Condition{
+				Type:     InvalidRegistryRoute,
+				Status:   True,
+				Reason:   RouteTestFailed,
+				Category: Critical,
+				Message:  fmt.Sprintf("Exposed registry route is invalid, response received : %#v", res),
```

Comment:
Let's change this message. `res` is going to be `nil` in the case of an error. put `err` here.

then below if it fails a status code check lets show the status code to the user

Reply:
Done.